src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -348,7 +392,16 @@ cpdef object compute_network(
 
             timestep += 1
 
-    return np.asarray(data_idx, dtype=np.intp), np.asarray(flowveldepth, dtype='float32')
+
+    # delete the duplicate results that shouldn't be passed along
+    # The upstream keys have empty results because they are not part of any reaches
+    # so we need to delete the null values that return
+    if len(fill_index_mask) > 0:
+        data_idx_ma = [ix for i, ix in enumerate(data_idx) if i not in fill_index_mask]
```

Comment:
Here we are reading data out of a numpy array into a list and then back into a numpy array. I wonder if it would be faster to construct a mask and then mask data_idx.

Reply:
```python
fill_mask = np.ones_like(data_idx)
fill_mask[list(fill_index_mask)] = 0
return np.asarray(data_idx[fill_mask], dtype=np.intp), np.asarray(flowveldepth_ma[fill_mask], dtype='float32')
```

Reply:
Looking at how fill_index_mask is constructed, we could construct the boolean mask directly I think.

Reply:
Will respond with #205